@re-/model/demos/generateCode/mapTSfiles.ts

```diff
@@ -1,56 +1,133 @@
+} = regex
 
-const getComment = (text: string, range: any) =>
-    text.slice(range.getPos(), range.getEnd())
+const codeDestinations = ["stackBlitz", "tests", "docs"]
 
-const stripCodeTag = (codeTag: string) =>
-    codeTag.replace("@re_place", "").trim()
+const breakdownGrabComment = (comment: string, name?: string) => {
+    const NAME_TAG = "-name:"
+    let optionsDestinations: string[] = []
+    let strippedComment: string
 
-const addFileContentsToMap = (
-    fileText: string,
-    astStatements: Statement<ts.Statement>[],
-    map: Record<string, string>
-) => {
-    for (const content of astStatements) {
-        const range = content.getLeadingCommentRanges()
-        if (range.length) {
-            const foundMatchingComment = getComment(fileText, range[0]).match(
-                tagMatch
+    const commentHasNameLine = comment.match(nameGrab)
+    const nameLine =
+        name ||
+        (!!commentHasNameLine
+            ? commentHasNameLine[0].replace(NAME_TAG, "").trim()
+            : `unnamedGrab${Math.floor(Math.random() * 100000)}`)
+
+    strippedComment = comment.replace(nameGrab, "").trim()
+
+    const endCommentMatch = comment.match(blockEndGrab)
+    if (!!endCommentMatch) {
+        strippedComment = strippedComment.replace(endCommentMatch[0], "")
+    }
+    const includeMatch = strippedComment?.match(includeGrab)
+    const excludeMatch = strippedComment?.match(excludeGrab)
+
+    try {
+        if (includeMatch && excludeMatch) {
+            throw new Error(
+                "Include and Exclude in the same grab comment is not supported."
             )
-            if (
-                foundMatchingComment &&
-                foundMatchingComment[0].split(" ").length > 2
-            ) {
-                const tag = stripCodeTag(foundMatchingComment[0])
-                map[tag] = content.getText()
-            }
         }
+        if (includeMatch) {
+            optionsDestinations = codeDestinations.filter((option) =>
+                strippedComment.toLowerCase().includes(option.toLowerCase())
+            )
+        } else if (excludeMatch) {
+            optionsDestinations = codeDestinations.filter(
+                (option) =>
+                    !strippedComment
+                        .toLowerCase()
+                        .includes(option.toLowerCase())
+            )
+        } else {
+            optionsDestinations = codeDestinations
+        }
+    } catch (e: any) {
+        console.error(e.message)
     }
+    return {
+        tagName: nameLine,
+        dest: optionsDestinations
+    }
+}
+const prepCommentThenMap = (
+    comment: string,
+    text: string,
+    mappedData: Record<string, object>
+) => {
+    const data = breakdownGrabComment(comment)
+    addBlockToMaps(data.dest, data.tagName, text, mappedData)
 }
+const mapBlocksAndStatements = (
+    file: SourceFile,
+    text: string,
+    mappedData: Record<string, object>
+) => {
+    let stack = new Stack<string>()
```

Comment:
Should be `const` here as well. In general, you only need to use let if you plan to reassign the whole value, not just e.g. push to an array or add a key to an object.

Also, let's call this `statementStack` so it's immediately obvious what it contains. I do like this approach to handling overlapping blocks a lot! 💪